openedx/features/partners/views.py

```diff
@@ -71,3 +102,250 @@ def reset_password_view(request):
             return JsonResponse({"Error": {"email": [response.content]}}, status=status.HTTP_403_FORBIDDEN)
         return response
     return JsonResponse({"Error": dict(reset_password_form.errors.items())}, status=status.HTTP_404_NOT_FOUND)
+
+
+class PartnerRegistrationView(RegistrationViewCustom):
```

Comment:
@mabdullahz as `RegistrationViewCustom` is a specialized type of `rest_framework.views.APIView` you can use serializers for validation and parsing of data in post and get methods. Please use serializer if it is possible otherwise we can go with the same approach.

Reply:
I moved this file from `partner/give2asia/views` to `partner/views` as the generalisation part of the ticket, and the changes are not mine. Also, we are using forms everywhere in all the registration views for validation and parsing of data. @zahid-arbisoft could give a better insight on this, if you want.

Reply:
@mabdullahz marking it resolved! thanks